bindings/python/iota_sdk/types/block/block.py

```diff
@@ -2,53 +2,60 @@
 # SPDX-License-Identifier: Apache-2.0
 
 from __future__ import annotations
-from enum import Enum
+from enum import Enum, IntEnum
 from dataclasses import dataclass
-from typing import List, Optional, Union, Dict
-from dacite import from_dict
+from typing import List, Optional
 from iota_sdk.types.common import HexStr, json
-from iota_sdk.types.payload import TaggedDataPayload, TransactionPayload
-from iota_sdk.utils import Utils
+
+
+class BlockType(IntEnum):
+    """Block types.
+
+    Attributes:
+        Basic (1): A Basic Block.
+        Validation (2): A Validation Block.
+    """
+    Basic = 1
+    Validation = 2
 
 
 @json
 @dataclass
 class Block:
-    """Represent the object that nodes gossip around the network.
+    """Base class for blocks.
+    """
+    type: int
+
+
+@json
+@dataclass
+class BlockMetadata:
+    """Block Metadata.
 
     Attributes:
-        protocol_version: The protocol version with which this block was issued.
+        block_id: The id of the block.
         strong_parents: Blocks that are strongly directly approved.
         weak_parents: Blocks that are weakly directly approved.
         shallow_like_parents: Blocks that are directly referenced to adjust opinion.
-        burned_mana: The amount of Mana the Account identified by the IssuerId is at most willing to burn for this block.
-        payload: The optional payload of this block.
+        is_solid: Whether the block is solid.
```

Comment:
These fields are wrong?

Reply:
you might not see the full context ... this is `BlockMetadat`.

Reply:
Yes, but BlockMetadata looks like this in 2.0
```rust
pub struct BlockMetadataResponse {
    pub block_id: BlockId,
    // TODO: verify if really optional: https://github.com/iotaledger/tips-draft/pull/24/files#r1293426314
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub block_state: Option<BlockState>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub tx_state: Option<TransactionState>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub block_failure_reason: Option<BlockFailureReason>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub tx_failure_reason: Option<TransactionFailureReason>,
}
```

Reply:
hmm, alright :+1:

Reply:
updated